src/python/WMCore/BossAir/Plugins/PyCondorPlugin.py

```diff
@@ -385,76 +385,79 @@ def submit(self, jobs, info=None):
         # unordered list of jobs with random sandboxes.
         # We intend to sort them by sandbox.
 
-        # We also need to do a secondary sort by number of cores
-        # because we can have jobs with different number of cores
-        # and the same sandbox and they cannot be submitted together.
+        # Do a secondary sort for numberOfCores and also for
+        # highIO jobs, they need to be submitted separately.
 
         submitDict = {}
         nSubmits   = 0
         for job in jobs:
             sandbox = job['sandbox']
             numberOfCores = job.get('numberOfCores', 1)
+            highIOjob = job.get('highIOjob', False)
             if not sandbox in submitDict.keys():
                 submitDict[sandbox] = {}
             if not numberOfCores in submitDict[sandbox]:
-                submitDict[sandbox][numberOfCores] = []
-            submitDict[sandbox][numberOfCores].append(job)
+                submitDict[sandbox][numberOfCores] = {}
+            if not highIOjob in submitDict[sandbox][numberOfCores]:
+                submitDict[sandbox][numberOfCores][highIOjob] = []
```

Comment:
There will always be a complexity of O(n^3) now, this is unavoidable. You cannot submit jobs in parallel that differ in sandbox, numberOfCores or the highIO flag. So you need to sort your jobs according to all three criteria. In practice it shouldn't be so bad though because the parameters your sort by are (sometimes highly) correlated. Like almost all highIO jobs are single core (except for multi core repacking in the Tier0, which is rare).